test/utils/test_config.jl

```diff
@@ -0,0 +1,104 @@
+module test_config
+
+using Test
+using JETLS
+
+const TEST_DICT = Dict(
+    "test_key1" => "test_value1",
+    "test_key2" =>  Dict(
+        "nested_key1" => "nested_value1",
+        "nested_key2" => Dict(
+            "deep_nested_key1" => "deep_nested_value1",
+            "deep_nested_key2" => "deep_nested_value2"
+        )
+    )
+)
+
+const TEST_DICT_DIFFERENT_VALUE = Dict(
+    "test_key1" => "newvalue_1",  # different value (reload required)
+    "test_key2" =>  Dict(
+        "nested_key1" => "nested_value1",
+        "nested_key2" => Dict(
+            "deep_nested_key1" => "newvalue_2", # different value (reload required)
+            "deep_nested_key2" => "newvalue_3"  # different value (reload not required)
+        )
+    )
+)
+
+const TEST_DICT_DIFFERENT_KEY = Dict(
+    "diffname_1" => "test_value1",  # different key name
+    "test_key2" =>  Dict(
+        "nested_key1" => "nested_value1",
+        "diffname_2" => Dict(       # different key name
+            "deep_nested_key1" => "deep_nested_value1",
+            "diffname_3" => "deep_nested_value2" # different key under a different key
+        )
+    ),
+)
+
+const TEST_RELOAD_REQUIRED = Dict(
+    "test_key1" => true,
+    "test_key2" => Dict(
+        "nested_key1" => false,
+        "nested_key2" => Dict(
+            "deep_nested_key1" => true,
+            "deep_nested_key2" => false
+        )
+    )
+)
+
+JETLS.is_reload_required_key(key_path::Vector{String}) =
+    JETLS.access_nested_dict(TEST_RELOAD_REQUIRED, key_path)
```

Comment:
Such overrides can cause serious invalidation, so they should be avoided as much as possible, even in tests.
```julia
global CONFIG_RELOAD_REQUIRED::Dict{String,Any} = ...
```
I think it would be better to set it up like this, and then wrap the test side with `try/finally` while replacing it with `JETLS.CONFIG_RELOAD_REQUIRED = ...`.